Library/Homebrew/api/cask.rb

```diff
@@ -23,12 +23,20 @@ def fetch_source(token, git_head: nil)
         sig { returns(Hash) }
         def all_casks
           @all_casks ||= begin
-            json_casks = Homebrew::API.fetch_json_api_file "cask.json",
-                                                           target: HOMEBREW_CACHE_API/"cask.json"
+            json_casks, updated = Homebrew::API.fetch_json_api_file "cask.json",
```

Comment:
Perhaps this logic could/should live in `Homebrew::API` given it's so similar for formulae and casks?